homeassistant/components/trafikverket_weatherstation/sensor.py

```diff
@@ -197,29 +209,34 @@ class TrafikverketWeatherStation(SensorEntity):
 
     def __init__(
         self,
-        weather_api,
-        name,
-        sensor_station,
+        weather_api: TrafikverketWeather,
+        name: str,
+        sensor_station: str,
         description: TrafikverketSensorEntityDescription,
-    ):
+    ) -> None:
         """Initialize the sensor."""
         self.entity_description = description
         self._attr_name = f"{name} {description.name}"
+        self._attr_unique_id = f"trafikverket_{name}_{description.key}"
         self._station = sensor_station
         self._weather_api = weather_api
-        self._weather = None
+        self._weather: WeatherStationInfo = None
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict:
         """Return the state attributes of Trafikverket Weatherstation."""
+        if self._weather.active:
```

Comment:
We may need to assert `self.weather` here to tell mypy if we know it will be set before the state update.

Reply:
Hi @MartinHjelmare . Don't understand what you mean by this one?

Reply:
`self.weather` is initialized as `None`. Mypy will complain unless we tell it that `self.weather` is not None anymore. If we know that tot be the case, we can do:
```suggestion
    assert self.weather is not None
        if self._weather.active:
```

Reply:
Thanks. The problem is that on a few occasions the initial setup fails to get these and therefore the sensor is not created at all. Using an assert here I guess would just make the same given it will raise and error and again the sensor is not created.
So I took a different approach to set default values instead which will be committed shortly.